store/sql_channel_store.go

```diff
@@ -405,7 +405,7 @@ func (s SqlChannelStore) get(id string, master bool, allowFromCache bool) StoreC
 				if metrics != nil {
 					metrics.IncrementMemCacheHitCounter("Channel")
 				}
-				result.Data = cacheItem.(*model.Channel)
+				result.Data = (cacheItem.(*model.Channel)).DeepCopy()
```

Comment:
How do you know that the object isn't being mutated while you're doing this copy? Wouldn't the cache need to do the copy while ensuring that no other threads are mutating the object?